test/skimNtuple2018_HHbtag.cpp

```diff
@@ -4273,168 +4276,71 @@ int main (int argc, char** argv)
-                  cout<<"ECON (E,Px,Py,Pz,M) "<<tlv_firstBjet.E()<<","<<tlv_firstBjet.Px()<<","<<tlv_firstBjet.Py()<<","<<tlv_firstBjet.Pz()<<","<<tlv_firstBjet.M()<<endl;
-                  cout<<"ECON B2"<<endl;
-                  cout<<"ECON (E,Px,Py,Pz,M) "<<tlv_secondBjet.E()<<","<<tlv_secondBjet.Px()<<","<<tlv_secondBjet.Py()<<","<<tlv_secondBjet.Pz()<<","<<tlv_secondBjet.M()<<endl;
-                  cout<<"ECON MET"<<endl;
-                  cout<<"ECON (E,Px,Py,Pz,M) "<<","<<ptmiss.Px()<<","<<ptmiss.Py()<<endl;
-                  cout<<"ECON METCOV "<<endl;
-                  cout<<"ECON "<<metcov (0,0)<<"  "<<metcov (0,1)<<endl;// = theBigTree.MET_cov00->at (chosenTauPair) ;
-                  cout<<"ECON "<<metcov (1,0)<<"  "<<metcov (1,1)<<endl;// = theBigTree.MET_cov10->at (chosenTauPair) ;
-                  cout<<"ECON tau1, tau2, b1, b2"<<endl;
-                  cout<<"ECON ";
-                  tlv_firstLepton.Print();
-                  cout<<"ECON ";
-                  tlv_secondLepton.Print();
-                  cout<<"ECON ";
-                  tlv_firstBjet.Print();
-                  cout<<"ECON ";
-                  tlv_secondBjet.Print();
-                  wrongHHK=true;
-                }
-              if(!wrongHHK)
-                {
-                  HHKmass = kinFits.getMH () ;
-                  HHKChi2 = kinFits.getChi2 () ;
-                }
-              else
-                {
-                  if(isOS)HHKmass = -333;
-                }
+          {
+            if(DEBUG)
+              {
+                 cout<<"---Kinfit Input debug---"<<endl;
+                 cout<<"L1"<<endl;
+                 cout<<"(pt,eta,phi,e) "<<tlv_firstLepton.Pt()<<","<<tlv_firstLepton.Eta()<<","<<tlv_firstLepton.Phi()<<","<<tlv_firstLepton.E()<<endl;
+                 cout<<"L2"<<endl;
+                 cout<<"(pt,eta,phi,e) "<<tlv_secondLepton.Pt()<<","<<tlv_secondLepton.Eta()<<","<<tlv_secondLepton.Phi()<<","<<tlv_secondLepton.E()<<endl;
+                 cout<<"B1"<<endl;
+                 cout<<"(pt,eta,phi,e,res) "<<tlv_firstBjet.Pt()<<","<<tlv_firstBjet.Eta()<<","<<tlv_firstBjet.Phi()<<","<<tlv_firstBjet.E()<<","<<bjet1_JER<<endl;
+                 cout<<"B2"<<endl;
+                 cout<<"(pt,eta,phi,e,res) "<<tlv_secondBjet.Pt()<<","<<tlv_secondBjet.Eta()<<","<<tlv_secondBjet.Phi()<<","<<tlv_secondBjet.E()<<","<<bjet2_JER<<endl;
+                 cout<<"MET"<<endl;
+                 cout<<"(Px,Py) "<<","<<ptmiss.Px()<<","<<ptmiss.Py()<<endl;
+                 cout<<"METCOV "<<endl;
+                 cout<<metcov(0,0)<<"  "<<metcov(0,1)<<endl;
+                 cout<<metcov(1,0)<<"  "<<metcov(1,1)<<endl;
+               }
+
+            KinFitter fitter(tlv_firstBjet, tlv_secondBjet, tlv_firstLepton, tlv_secondLepton, ptmiss, stableMetCov, bjet1_JER, bjet2_JER);
+            KinFitter fitterRaw(tlv_firstBjet, tlv_secondBjet, tlv_firstLepton, tlv_secondLepton, ptmiss, stableMetCov, bjet1_JER, bjet2_JER);
```

Comment:
Just a comment: at some poit we should get rid of this fitterRaw since it's exactly the same there is no need of doing it twice, actually we could save some time/cpu.
For now it's fine like this, it's just a reminder for the future.

Reply:
Yes, I never understood the "raw fit" actually, I kept it just for consistency. I suggest I just remove it now and we are done with it.